Clarify wording for Learn More screen in Uncertain Delivery State

marionbarker opened the pull request:
Here's a first cut for modifying the Learn More screen for DASH pods.

The top part of the graphic is what is displayed now.
The bottom part matches the current commit.

For both cases, I had to scroll up to read all the text.

Removed graphic because I updated words based on comments. See later comment for new graphics.

marionbarker committed:
Clarify wording for Learn More screen when Unable to Reach Pod

motinis commented on OmniBLE/PumpManagerUI/Views/DeliveryUncertaintyRecoveryView.swift:
```diff
@@ -19,7 +19,7 @@ struct DeliveryUncertaintyRecoveryView: View {
     var body: some View {
         GuidePage(content: {
-            Text(String(format: LocalizedString("%1$@ has been unable to communicate with the pod on your body since %2$@.\n\nWithout communication with the pod, the app cannot continue to send commands for insulin delivery or display accurate, recent information about your active insulin or the insulin being delivered by the Pod.\n\nMonitor your glucose closely for the next 6 or more hours, as t ...
+            Text(String(format: LocalizedString("%1$@ has been unable to communicate with the pod on your body since %2$@.\n\nWithout communication with the pod, the app cannot resolve information about your active insulin or the insulin being delivered by the Pod. If you cannot resolve communication with this pod, you should monitor your glucose closely for the next 6 or more hours, as there may ...
```
Typo here - `Deactive Pod` should be `Deactivate Pod`.

Also it seems it would be a good idea to provide some additional guidance here - what is the minimum amount of time someone should wait? I think the statement "you should have waited longer" isn't helpful... to me it speaks to not actually wanting to deactivate the pod - rather it is "disconnect from pod" or "unpair pod" (and as part of that it would check the comms with the pod one last time and only disconnect if it fails).

marionbarker committed:
wording change for Learn More screen when Unable to Reach Pod

marionbarker commented:
Thanks for the input.

Edited to remove graphics on 25 January 2024.
Changes words for consistency with https://github.com/LoopKit/OmniKit/pull/23

motinis commented:
Looks great - the PR on the slider needs to include this Deactivate Pod button too

culdeus commented:
Can I ask why there needs to be this deactivate pod button here at all.  If the pod has lost comm, how do you deactivate it from this menu?

ps2 commented:
Yes, it should skip the "deactivate" part, and just have a "Forget Pod" button.  There does need to be a way to get out of this state, if comms cannot eventually be restored.

marionbarker committed:
add a Keep Waiting primary button, rename destructive button, modify words

marionbarker commented:
Making changes one step at a time.

This commit
* adds a primary button with the label "Keep Waiting", which dismisses this screen
* renames the destructive button from "Deactivate Pod" to "Discard Pod"
* no logic change was made to the action taken by that destructive button

Edited to remove graphics on 25 January 2024.
Changed words for consistency with https://github.com/LoopKit/OmniKit/pull/23

marionbarker commented:
I will need guidance if additional modifications are desired to the actions. Right now, the next screen after tapping Discard Pod (on the screen in prior comment) is the Deactivate Pod screen, which, if comms still fail, takes you to the Discard Pod screen as shown in the graphic below.

![pr-104-unmodified-deactivate-then-discard-screen](https://github.com/LoopKit/OmniBLE/assets/19607791/ea3c1f89-6d97-4c1d-ac60-63c3c2bbac78)

ps2 reviewed (approved):
LGTM! Thanks!

marionbarker commented:
See comment when testing similar change for OmniKit: https://github.com/LoopKit/OmniKit/pull/23#issuecomment-1888537322

marionbarker committed:
modify delivery uncertainty words to be consistent with OmniKit

marionbarker commented:
Updated with the graphics that match the latest commit (25 January 2024). After working with the uncertain comms for OmniKit, I modified those words then came back to this PR and modified them for consistency.

Top row - comes from LoopKit and is not modified.
Middle row - shows top and bottom of the screen addressed by this PR
Bottom row - shows the screen shown after comms are restored (not modified by this PR)


![omnible-pr-104](https://github.com/LoopKit/OmniBLE/assets/19607791/3b63a447-9901-4ff4-bd88-16187b93a07d)

ps2 commented on OmniBLE/PumpManagerUI/Views/DeliveryUncertaintyRecoveryView.swift:
```diff
@@ -19,16 +19,23 @@ struct DeliveryUncertaintyRecoveryView: View {
                     self.model.podDeactivationChosen()
                 }) {
-                    Text(LocalizedString("Deactivate Pod", comment: "Button title to deactive pod on uncertain program"))
+                    Text(LocalizedString("Discard Pod", comment: "Button title to discard pod on uncertain program"))
```
I think this still takes the user to deactivate, rather than discard, but I'm on the fence about which is better.

marionbarker replied:
Yes it does because I couldn’t figure out how to change it. 

But on the other hand, if it succeeds in reaching the pod, the pod gets deactivated so there’s no screamer but at the same time you’re mad because you gave up too soon.
